tmt/steps/provision/local.py

```diff
@@ -133,6 +133,34 @@ def pull(
             extend_options: Optional[list[str]] = None) -> None:
         """ Nothing to be done to pull workdir """
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """fetch and return content of a requested log"""
+        if log_name == 'dmesg':
+            log_content = self.execute(Command('dmesg')).stdout
+        if log_content:
+            return log_content
+        return None
+
+    def store_log(self, log_name: str, log_content: str) -> Optional[str]:
```

Comment:
The file path to save into must be the parameter: it's not up to the guest instance to decide where the log is supposed to end. It may be a file in step's workdir, it may be a file on NFS mount, or it may be `/dev/null`. The point is, the guest does not know the context. Accept file path and content, and save. Plus, there is not much to return from this method, `-> None` would be enough. Your code returns the path, but that should be an input rather than outcome of the call to `store_log()`.